### PR TITLE
feat: automatically release cyctl using goreleaser

### PR DESCRIPTION
**Pull request by SpiffyEight77**
## 📑 Description

issue #192

This is the minimal setup for packaging cyctl using goreleaser. I'm not sure if goreleaser's changelog will overwrite the existing draft release content, so I recommend testing it through a prerelease first to ensure foolproof results.

For more information, please visit this [release](https://github.com/SpiffyEight77/cyclops/releases/tag/v0.6.4).

## ✅ Checks
- [x] I have updated the documentation as required
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by petar-cvit** [comment]
Hey @SpiffyEight77, thanks for the PR. I checked it out but I will review it in the following couple of days.

> I'm not sure if goreleaser's changelog will overwrite the existing draft release content

I checked your repo and the releases made there, and I see only the `full changelog` on the release. Could you create and merge a (empty) PR into your `feat/gha-goreleaser` and see if the changelog will be there with your merged PR.
If I merge this PR to cyclops:main, I don't want it to break future releases

**Comment by SpiffyEight77** [comment]
@petar-cvit 
> I checked your repo and the releases made there, and I see only the full changelog on the release. Could you create and merge a (empty) PR into your feat/gha-goreleaser and see if the changelog will be there with your merged PR.
If I merge this PR to cyclops:main, I don't want it to break future releases

Okay, I will run some tests to see if the changelog will be chenged or not in my `feat/gha-goreleaser`.

**Comment by SpiffyEight77** [comment]
After a lot of testing, the changelog would not be changed by the goreleaser.

Here are the results of the auto [release](https://github.com/SpiffyEight77/cyclops/releases/tag/v0.6.10) and [pre-release](https://github.com/SpiffyEight77/cyclops/releases/tag/v0.6.9-rc.2).

Please take a look at for these when you are available.

File: cyctl/.goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      - amd64
+      - arm64
+    flags:
+      - -trimpath
+    ldflags:
+      - -s -w -X github.com/cyclops-ui/cyclops/common.CliVersion={{.Version}}
+
+archives:
+  - format: tar.gz
+    name_template: "cyctl_{{ .Os }}_{{ .Arch }}"
```
Comment (petar-cvit) [inline cyctl/.goreleaser.yaml:51]: @SpiffyEight77  Can we include the version to the name? Same goes for nfpm

Comment (SpiffyEight77) [reply cyctl/.goreleaser.yaml:51]: > @SpiffyEight77 Can we include the version to the name? Same goes for nfpm

@petar-cvit

Sure, we can add the version to the name like `cyctl_darwin_arm64_v0.6.2.tar.gz` or use any other naming format you prefer.

I'll handle this later tonight after work.

Comment (petar-cvit) [reply cyctl/.goreleaser.yaml:51]: Yeah, this looks good, thanks

Comment (SpiffyEight77) [reply cyctl/.goreleaser.yaml]: The version is included in the file name. For more details, please visit the [latest release](https://github.com/SpiffyEight77/cyclops/releases/tag/v0.6.11) in the forked repository.

**Comment by petar-cvit** [comment]
Hey @SpiffyEight77, now that you have set up brew for cyctl, can we add `brews` to goreleaser? Do you want to do it in a separate PR or add it to this one?

**Comment by SpiffyEight77** [comment]
> Hey @SpiffyEight77, now that you have set up brew for cyctl, can we add `brews` to goreleaser? Do you want to do it in a separate PR or add it to this one?

The brews in `gorelease` are for `homebrew-tap.` If we want to maintain a third-party version for `cyctl`, we might need to create a new repository named `cyclops/homebrew-tap`, similar to [terraform](https://github.com/hashicorp/homebrew-tap).

The `cyctl` version in `homebrew-core` will automatically update when a new release is available.

**Comment by SpiffyEight77** [comment]
> Hey @SpiffyEight77, now that you have set up brew for cyctl, can we add `brews` to goreleaser? Do you want to do it in a separate PR or add it to this one?

@petar-cvit 

The brews in `gorelease` are for `homebrew-tap.` If we want to maintain a third-party version for `cyctl`, we might need to create a new repository named `cyclops/homebrew-tap`, similar to [HashiCorp](https://github.com/hashicorp/homebrew-tap).

**Review by petar-cvit — APPROVED** [review]
Thanks a lot @SpiffyEight77 🧡

**Pull request merged**